src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -413,37 +412,39 @@ class spell_pal_divine_storm : public SpellScriptLoader
         {
             PrepareSpellScript(spell_pal_divine_storm_SpellScript);
 
-        public:
-            spell_pal_divine_storm_SpellScript()
-            {
-                healPct = 0;
-            }
-
-        private:
-            uint32 healPct;
-
             bool Validate(SpellInfo const* /*spellInfo*/) override
             {
-                if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STORM_DUMMY))
+                if (!sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STORM_DAMAGE) ||
+                    !sSpellMgr->GetSpellInfo(SPELL_PALADIN_DIVINE_STORM_DUMMY))
                     return false;
                 return true;
             }
 
-            bool Load() override
-            {
-                healPct = GetSpellInfo()->GetEffect(EFFECT_1)->CalcValue(GetCaster());
-                return true;
-            }
-
-            void TriggerHeal()
+            void HandleOnCast()
             {
                 Unit* caster = GetCaster();
-                caster->CastCustomSpell(SPELL_PALADIN_DIVINE_STORM_DUMMY, SPELLVALUE_BASE_POINT0, (GetHitDamage() * healPct) / 100, caster, true);
+
+                std::list<Unit*> targetList;
+                float radius = GetSpellInfo()->GetEffect(EFFECT_0)->CalcRadius(GetCaster());
+
+                Trinity::NearestAttackableUnitInObjectRangeCheck u_check(caster, caster, radius);
+                Trinity::UnitListSearcher<Trinity::NearestAttackableUnitInObjectRangeCheck> searcher(caster, targetList, u_check);
+                caster->VisitNearbyObject(radius, searcher);
```

Comment:
Hm? Because this spell it's only for one target. We need casting 1 spell per target (sniff src)